pkg/metrics/filters/filters.go

```diff
@@ -0,0 +1,120 @@
+package filters
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apiserver/pkg/authentication/authenticatorfactory"
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+	"k8s.io/apiserver/pkg/authorization/authorizerfactory"
+	"k8s.io/apiserver/pkg/server/options"
+	authenticationv1 "k8s.io/client-go/kubernetes/typed/authentication/v1"
+	authorizationv1 "k8s.io/client-go/kubernetes/typed/authorization/v1"
+	"k8s.io/client-go/rest"
+	metricsserver "sigs.k8s.io/controller-runtime/pkg/metrics/server"
+)
+
+// WithAuthenticationAndAuthorization provides a metrics.Filter for authentication and authorization.
+// Metrics will be authenticated (via TokenReviews) and authorized (via SubjectAccessReviews) with the
+// kube-apiserver.
+// For the authentication and authorization the controller needs a ClusterRole
+// with the following rules:
+// - apiGroups:
+//   - authentication.k8s.io
+//     resources:
```

Comment:
Indent is wrong on `resources` and `verbs` I think, same for the samples below

Reply:
You're right. Seems to be done by go format somehow. I'll see how I can fix it

Reply:
I couldn't find anything that would make it possible to have basically inline YAML here. Seems like godoc always treats it as a list.

I changed it to something which is clearly not a YAML snippet but still documents the required rules